src/rust/protocol/thrift/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+//! A protocol crate for Thrift binary protocol.
+
+use protocol_common::Compose;
+use protocol_common::Parse;
+use protocol_common::{ParseError, ParseOk};
+use std::io::Write;
+
+/// An opaque Thrift message
+pub struct Message {
+    data: Box<[u8]>,
+}
+
+#[allow(clippy::len_without_is_empty)]
+impl Message {
+    pub fn len(&self) -> usize {
+        self.data.len()
+    }
+}
+
+impl Compose for Message {
+    fn compose(&self, session: &mut session::Session) {
+        let _ = session.write_all(&(self.data.len() as u32).to_be_bytes());
+        let _ = session.write_all(&self.data);
+    }
+}
+
+/// A parser which retrieves the bytes for a complete Thrift message.
+#[derive(Clone)]
+pub struct MessageParser {
```

Comment:
Since we have a single parser that works in both directions, I'm going with `messages_parsed` and `messages_composed`. This can be overhauled as part of the stats work we've been discussing in #446